tests/evm/test_create.py

```diff
@@ -305,59 +315,63 @@ def test_create_create2(
- 
-        # callee's call context
+        # Write contract's nonce to 1
         rw_dictionary \
-            .call_context_read(next_call_id, CallContextFieldTag.CallerId, CURRENT_CALL_ID) \
-            .call_context_read(next_call_id, CallContextFieldTag.TxId, 1) \
-            .call_context_read(next_call_id, CallContextFieldTag.Depth, stack_depth+1) \
-            .call_context_read(next_call_id, CallContextFieldTag.CallerAddress, Word(caller.address)) \
-            .call_context_read(next_call_id, CallContextFieldTag.CalleeAddress, Word(contract_address)) \
-            .call_context_read(next_call_id, CallContextFieldTag.IsSuccess, is_success) \
-            .call_context_read(next_call_id, CallContextFieldTag.IsStatic, is_static) \
-            .call_context_read(next_call_id, CallContextFieldTag.IsRoot, False) \
-            .call_context_read(next_call_id, CallContextFieldTag.IsCreate, True) \
-            .call_context_read(next_call_id, CallContextFieldTag.CodeHash, Word(EMPTY_CODE_HASH))
+            .account_write(contract_address, AccountFieldTag.Nonce, 1, 0)
+        
+        if has_init_code:
+            # copy_table
+            copy_circuit = CopyCircuit().copy(
+                randomness_keccak,
+                rw_dictionary,
+                CURRENT_CALL_ID,
+                CopyDataTypeTag.Memory,
+                init_codes_hash,
+                CopyDataTypeTag.Bytecode,
+                stack.offset,
+                stack.offset + stack.size,
+                0,
+                stack.size,
+                src_data,
+            )
+
+            # caller's call context
+            rw_dictionary \
+                .call_context_write(CURRENT_CALL_ID, CallContextFieldTag.ProgramCounter, next_program_counter) \
+                .call_context_write(CURRENT_CALL_ID, CallContextFieldTag.StackPointer, 1023) \
+                .call_context_write(CURRENT_CALL_ID, CallContextFieldTag.GasLeft, caller_gas_left) \
+                .call_context_write(CURRENT_CALL_ID, CallContextFieldTag.MemorySize, next_memory_size) \
+                .call_context_write(CURRENT_CALL_ID, CallContextFieldTag.ReversibleWriteCounter, caller_ctx.reversible_write_counter + 1) 
     
+            # callee's call context
+            rw_dictionary \
+                .call_context_read(next_call_id, CallContextFieldTag.CallerId, CURRENT_CALL_ID) \
+                .call_context_read(next_call_id, CallContextFieldTag.TxId, 1) \
+                .call_context_read(next_call_id, CallContextFieldTag.Depth, stack_depth+1) \
+                .call_context_read(next_call_id, CallContextFieldTag.CallerAddress, Word(caller.address)) \
+                .call_context_read(next_call_id, CallContextFieldTag.CalleeAddress, Word(contract_address)) \
+                .call_context_read(next_call_id, CallContextFieldTag.IsSuccess, is_success) \
+                .call_context_read(next_call_id, CallContextFieldTag.IsStatic, is_static) \
```

Comment:
Are line 178 and 179 still true (you removed the check in create.py)?
`    # can't be a static all
    is_static = 0`
If not, does it make sense to add a test with is_static = 1 ?

Reply:
Yes, L178 and L179 are true, it can't be static in CREATE/CREATE2 context. I accidentally removed the constraint in create.py bcs I was thinking it has been done [here](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/466/files#diff-5cd2ba2f17f8ac650dee9a0e2f167f35824d025dde24eaa047c8ebb72d1c6b31R196). However, I was wrong bcs that's for next call context instead of the current one.

Nice catch! Fixed in [c4fe5d1](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/466/commits/c4fe5d1eed01b51f056c8263acd16b73861aada8). Thanks!